### ISSUE TITLE
[0.77] Support view transitions on Android

### ISSUE THREAD

**Issue by kkafar** [reactwg/react-native-releases#675]
### Target Branch(es)

0.77-stable

### Link to commit or PR to be picked

https://github.com/facebook/react-native/commit/f402ed17fa6d75aea24e2ad99a8b8d8ad20840e3

### Description

The commit is well described (see the PR here: https://github.com/facebook/react-native/pull/47634, but note that @tdn120 introduced few changes before merging that are not reflected in the PR, but are present in the commit I've pasted above ☝🏻 )

Basically the commit adds support for view transitions & is crucial for popular community library `react-native-screens` to work on Fabric. It would be great to have it picked quickly, so that it is present in as many RCs as possible and can be battle-tested by others.
